plugins/imjournal/imjournal.c

```diff
@@ -520,24 +520,32 @@ persistJournalState(void)
 	DEFiRet;
 	FILE *sf = NULL; /* state file */
 	char tmp_sf[MAXFNAME];
-	size_t n;
 
 	DBGPRINTF("Persisting journal position, cursor: %s, at head? %d\n",
 			  journalContext.cursor, journalContext.atHead);
 
 	/* first check that we have valid cursor */
 	if (!journalContext.cursor) {
+		DBGPRINTF("Journal cursor is not valid, ok...\n");
 		ABORT_FINALIZE(RS_RET_OK);
 	}
 
 	/* we create a temporary name by adding a ".tmp"
 	 * suffix to the end of our state file's name
+	 *
+	 * we use snprintf() to safely honor the boundaries
+	 * of the temporary state file name buffer by using
+	 * a precision specifier, which will limit the number
+	 * of bytes taken from cs.stateFile to what will fit
```

Comment:
We do not need to take care of this - snprintf already does. If the buffer is too slow, snprintf returns a size larger than the buffer **without doing any writes**. This is what the previous code uses, which IMHO is perfectly OK (but I may be wrong, pls point out where exactly if so).

Reply:
Hello @rgerhards - The new snprintf call writes the file name and extension all in one go. The comment is explaining the use of the precision specifier to truncate just the cs.stateFile part of the string, and avoid the need for the alternate strncpy() strncat() block. Are you asking for the comment to change?

Reply:
looks like I had a bad day during the review. Makes sense.